coala_quickstart/generation/Settings.py

```diff
@@ -0,0 +1,99 @@
+import os
+from collections import OrderedDict
+
+from pyprint.NullPrinter import NullPrinter
+from coalib.output.printers.LogPrinter import LogPrinter
+from coala_utils.Extensions import exts
+from coala_quickstart.generation.Utilities import (
+    split_by_language, get_extensions, get_gitignore_glob)
+from coalib.settings.Section import Section
+from coalib.output.ConfWriter import ConfWriter
+from coalib.collecting.Collectors import collect_files
+
+
+def generate_settings(project_dir, project_files, relevant_bears):
+    """
+    Generates the settings for the given project.
+
+    :param project_dir:
+        Full path of the user's project directory.
+    :param project_files:
+        A list of file paths matched in the user's project directory.
+    :param relevant_bears:
+        A dict with language name as key and bear classes as value.
+    :return:
+        A dict with section name as key and a ``Section`` object as value.
+    """
+    lang_map = {lang.lower(): lang for lang in relevant_bears}
+    lang_files = split_by_language(project_files)
+    extset = get_extensions(project_files, lang_map)
+
+    null_printer = LogPrinter(NullPrinter())
+    gitignore_globs = get_gitignore_glob(project_dir)
+
+    settings = OrderedDict()
+
+    default = Section("default", None)
+    default["bears"] = ", ".join(
+        bear.name for bear in relevant_bears[lang_map["all"]])
+
+    extensions_used = [ext[1:]
+                       for lang in lang_files
+                       for ext in extset[lang]]
+    default["files"] = "**.(" + "|".join(extensions_used) + ")"
+    all_files = set()
+
+    for lang in lang_files:
+        all_files.update(set(collect_files(
+            [os.path.join(project_dir, "**" + ext) for ext in extset[lang]],
+            null_printer)))
+
+    ignores = [".git/**"]
+    for glob in gitignore_globs:
+        gitignore_files = {file
+                           for file in collect_files([glob], null_printer)}
+        if not all_files.isdisjoint(gitignore_files):
+            ignores.append(os.path.relpath(glob, project_dir))
+    default["ignore"] = ", ".join(ignores)
+
+    settings["default"] = default
+
+    for lang in lang_files:
+        if lang == "unknown" or lang == "all":
+            continue
+
+        section = Section(lang, None)
+
+        section["files"] = ", ".join("**" + ext for ext in extset[lang])
+
+        section["bears"] = ", ".join(
+            bear.name for bear in relevant_bears[lang_map[lang]])
+
+        settings[lang_map[lang]] = section
+
+    return settings
```

Comment:
I think we have some duplication here, generating the default section isn't much different from the others is it? Can you make a function generate_section that generates files, bears, etc. and patch in the ignore in the default section later?